modules/tinymce/src/plugins/media/main/ts/core/UpdateHtml.ts

```diff
@@ -161,7 +161,7 @@ const updateHtml = (html: string, data: Partial<MediaData>, updateAll?: boolean)
               const attrs: any = [];
               attrs.map = {};
 
-              if (sourceCount < index) {
```

Comment:
This seems like the correct fix though and it seems to have broken in https://github.com/tinymce/tinymce/commit/1fc08c7ecdc2faab9d3a299be3297c907e1375ed when the offsets where changed.